### ISSUE TITLE
Remove dependency on Google Play services

### ISSUE THREAD

**Issue by algazarra** [mycelium-com/wallet#162]
![mycelium-gapps-small](https://cloud.githubusercontent.com/assets/7598320/7255765/4d935828-e843-11e4-9252-f119d1395a78.png)

I just updated Mycelium and unlike previous versions, it now depends on Google Play services. Considering Mycelium's focus on anonimity and security, requiring shady/proprietary GApps seems a terrible strategic choice. 

As said by MaR-V-iN from [NoGAPPS](http://forum.xda-developers.com/showthread.php?s=a7bf27eb98e3bcefb7e58fb46d09710b&t=1715375):
> gapps track users more than most other apps as they are system-apps and therefore got full rights on android.

There are dozens of ROMs which ship without GApps, such as CyanogenMod, Omni and Replicant. I believe there is no valid technical justification to use GApps and force all these users to install them.

**Comment by spottedmarley** [comment]
here here!

**Comment by disposable-ksa98** [comment]
+1 Mycelium pls

**Comment by vivaserver** [comment]
+1

**Comment by xinobrax** [comment]
+1

**Comment by Kirvx** [comment]
+1

**Comment by BItim** [comment]
+2

**Comment by dllud** [comment]
+1

**Comment by jandreske** [comment]
From which version did you update?

-------- Ursprüngliche Nachricht --------
Von: David Ludovino <notifications@github.com> 
Datum: 22.04.2015  01:34  (GMT+01:00) 
An: mycelium-com/wallet <wallet@noreply.github.com> 
Betreff: Re: [wallet] (!) Remove dependency on Google Play services (#162) 

+1

**Comment by algazarra** [comment]
@jandreske can't say for sure. But I installed it less than 1 year ago.

**Comment by Giszmo** [comment]
Yep, [that](https://github.com/mycelium-com/wallet/blob/master/public/mbw/src/main/AndroidManifest.xml#L10) looks like a dependency you won't fulfill on non-google devices.

I see how it's convenient but I totally support to not only ask what's the most convenient option.

**Comment by Giszmo** [comment]
I just tried and I guess this is resolved. While evil google blobs might pose a security risk on evil google phones, as far as I can see these evil google blobs are not packaged with the actual apk, so on an all open source android phone, users should be fine.

**Comment by dllud** [comment]
Sure, Google blobs are not packaged with the apk, though they seem to be needed by some Local Trader function. The goal of this issue is to remove such requirement and get everything working without Google blobs.

I just downloaded the latest apk ([mbw-prodnet-release-2.5.8build25806.apk](https://www.mycelium.com/download/apk/mbw-prodnet-release-2.5.8build25806.apk)) and the "Get Google Play services" warning still appears when I open Local Trader. Though, it seems everything is working after dismissing it. I wonder if the warning can be safely removed.

**Comment by nighthawk24** [comment]
I've forked the latest v2.6 and removed the dependency on play services https://github.com/nighthawk24/wallet

WARNING: This is for my personal use only and I would not store any significant amounts on a mobile wallet.

**Comment by dllud** [comment]
@nighthawk24 Perhaps you can turn it into a merge request here on the main repo?

**Comment by Giszmo** [comment]
I guess it's difficult to maintain this branch and I doubt we would merge it (unless Google bans all bitcoin apps or something). For reference, sure, go ahead and do a merge request. If it had good community support I could also imagine to see it maintained in a branch but it would probably not get much attention by the Mycelium devs which would make such an endorsement difficult.

**Comment by Rassah** [comment]
Dropping google would require us to have our own Push system, and switch to using OpenStreetMap. Both are somewhat large projects, but maybe we'll keep those goals in mind should we finally get around to updating LocalTrader.

**Comment by nighthawk24** [comment]
@dllud Removing the play services dependency breaks the notification functionality. I do not feel it is okay to create a pull request in master branch.

@Giszmo Is there a way you could create a new branch for non-play-services requirement, and I can make a pull request there.

**Comment by nighthawk24** [comment]
@Rassah In the future, would it be a good idea to separate LocalTrader from the core wallet?

**Comment by Giszmo** [comment]
@nighthawk24 any pull request that would leave the users without notification would not get pulled. Not even to a new branch. But doing a pull request here, would create a point for discussion for those that want to work on it. Once you have your bitmessage notifications or whatever alternative working.

Regarding LT as a separate module, yes, in the upcoming wallet things will be more modular and I would prefer to have it in separate apps, so that the "buy/sell bitcoins" button checks if LT is installed and if not, gets you to the playstore or else, it would start it. This would reduce the complexity and permissions for the core app and allow to have alternative LT modules.
